### DISCUSSION TITLE
CLI: expose `NX_BAIL` as env variable in order to configure `--nxBail`

### DISCUSSION THREAD

**Discussion by Hotell** [nrwl/nx#26899] [Feature Requests]
To follow nx CLI configuration setup via [global env variables](https://nx.dev/reference/environment-variables) it would be really helpful to add variable to set `--nxBail` argument.

we want this behaviour in all our pipelines and ATM it adds unnecessary churn to the whole CI setup ([example](https://github.com/microsoft/fluentui/pull/31949/files#diff-2885e20595f8bf2dde10797f4f08ec854150fc783b2f2b4c1fea57d69fdd9aa0)

**Current:**

will bail on 1st failed `build` target execution

`nx affected -t b --nxBail`

**Expected:**

```yml
## ADO pipeline config
variables:
  NX_BAIL: true
```

will bail on 1st failed `build` target execution

`nx affected -t build`

**Comment by 100terres** [comment]
That would be a super useful environment variable.

I'm working on a project with many commands that use `--nx-bail`. We sometimes want to run everything even on failure, but it's a bit of a pain to edit the GitHub action workflow file and remove this flag from the different commands.

To make this easy, we've added a `workflow_dispatch` event with an input that lets us disable the bail flag and run everything without changing the workflow file. However, since we can't set an environment, our config is a bit complex. 

It would greatly simplify our setup if we could do the following:

```yaml
env:
  NX_BAIL: ${{ inputs.activate_nx_bail }}
```

**Comment by jase88** [comment]
https://github.com/nrwl/nx/pull/34711
